docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,304 @@
+
+```go
+type LightClientAttackEvidence struct {
+  ConflictingBlock *LightBlock
+  CommonHeight  int64
+  Timestamp  time.Time
+  Addresses []Address // we need this data in order to know which validators were at fault.
+}
+```
+
+The addresses would also be used as a look up table such that we can store evidence per address.
+
+
+### Batch Framework
+
+The last approach of this category would be to consider batch only evidence. This works fine with `LightClientAttackEvidence` but would require alterations to `DuplicateVoteEvidence` which would most likely mean that the consensus would send conflicting votes to a buffer in the evidence module which would then wrap all the votes together per height before gossiping them to other nodes and trying to commit it on chain. At a glance this may improve IO and verification speed and perhaps more importantly grouping validators gives the application and Tendermint a better overview of the severity of the attack. However individual evidence has the advantage that it is easy to check if a node already has that evidence meaning we just need to check hashes to know that we've already verified this evidence before. Batching evidence would imply that each node may have a different combination of duplicate votes which may complicate things.
+
+
+#### An example of a possible implementation path
+
+`LightClientAttackEvidence` won't change but the evidence interface will need to look like the proposed one above and `DuplicateVoteEvidence` will need to change to encompass multiple double votes. A problem with batch evidence is that it needs to be unique to avoid people from submitting different permutations.
+
+> I most likely have not captured the full picture of all these approaches and therefore would encourage others to add their takes on each of them.
+
+
+## Evidence Interface
+
+*Note: any changes to the interface will be changing the block data structure and is hence breaking*
+
+As already been alluded to it is most likely that we will need to remove `Verify()` as it's difficult to define a set of arguments that will satisfy the requirements for both duplicate vote verification and light client attack verification (not to mention future evidence and their requirements).
+
+Depending on the composition we might need to change `Address()` to accomodate either individual and/or batch evidence. It may also be a good idea to have internal fields that the evidence pool populates such as time and validators so modules outside of the evidence pool (specifically the `BlockExecutor` don't need to query for more resources)
+
+`Equal()` can also be seen as redundant if we set `Hash()` in the right way that we can just compare evidence hashes for uniqueness. We already use hashes for database lookups
+
+`Type()` could be a new method which would be helpful in constructing abci evidence
+
+`ToABCI()` is another alternative that was written as a TODO in the actual code base.
```

Comment:
What would this function do? Turn an `Evidence` into whatever type we pass over ABCI?

Reply:
Yup it was a note made in the codebase that I thought could be one approach. Actually what I finally propose is something similar where instead of the block executor creating the abci evidence the evidence pool does